honeybee_radiance_folder/folder.py

```diff
@@ -0,0 +1,584 @@
+
+    @property
+    def grid_folder(self):
+        """Sensor grids folder."""
+        return self.GRID
+
+    def grid_files(self, rel_path=True):
+        """List of sensor grid files."""
+        return self._find_files(self.GRID, '.pts', rel_path)
+
+    @property
+    def view_folder(self):
+        """View files folder."""
+        return self.VIEW
+
+    def view_files(self, rel_path=True):
+        """List of view files."""
+        return self._find_files(self.VIEW, '.vf', rel_path)
+
+    def write(self, overwrite=True):
+        """Write model folder."""
+        for subfolder in self.ASSET:
+            directory = os.path.join(self.folder, subfolder)
+            if os.path.exists(directory) and not overwrite:
+                raise ValueError('{} already exist.'.format(directory))
+            os.makedirs(directory)
+
+
+class OutputFolder(_FolderCollection):
+    """Radiance output folder.
+
+    ├───output                    [required]
+        ├───matrix
+        ├───octree
+        ├───postprocess
+        └───temp
+    """
+    OUTPUT_ROOT = 'output'
+    OCTREE = 'output/octree'
+    MATRIX = 'output/matrix'
+    TEMP = 'output/temp'
+    POSTPROCESS = 'output/postprocess'
+    OUTPUT = OUTPUT_ROOT, OCTREE, MATRIX, TEMP, POSTPROCESS
+    REQUIRED = (OUTPUT_ROOT,)
+
+    @property
+    def root(self):
+        """Output root folder."""
+        return self.OUTPUT_ROOT
+
+    @property
+    def octree(self):
+        """Folder for octree files."""
+        return self.OCTREE
+
+    @property
+    def matrix(self):
+        """Matrix outputs folder."""
+        return self.MATRIX
+    
+    @property
+    def temp(self):
+        """Temporary files folder."""
+        return self.TEMP
+    
+    @property
+    def postprocess(self):
+        """Postprocess folder."""
+        return self.postprocess
+
+    def write(self, overwrite=True):
+        """Write model folder."""
+        for subfolder in self.OUTPUT:
+            directory = os.path.join(self.folder, subfolder)
+            if os.path.exists(directory) and not overwrite:
+                raise ValueError('{} already exist.'.format(directory))
+            os.makedirs(directory)
+
+
+class SystemFolder(_FolderCollection):
+    """System folder includes settings for runs like options, tasks.json."""
+    SYSTEM_ROOT = 'system'
+    SYSTEM = (SYSTEM_ROOT,)
+    REQUIRED = (SYSTEM_ROOT,)
+
+    @property
+    def root(self):
+        """System root folder."""
+        return self.SYSTEM_ROOT
+
+    def write(self, overwrite=True):
+        """Write model folder."""
+        for subfolder in self.SYSTEM:
+            directory = os.path.join(self.folder, subfolder)
+            if os.path.exists(directory) and not overwrite:
+                raise ValueError('{} already exist.'.format(directory))
+            os.makedirs(directory)
+
+
+class Folder(object):
+    """Radiance folder structure."""
+
+    __version__ = '2.0.0'  # radiance folder version
```

Comment:
Hm.  Do you think you'll revise this again? I'm just curious about how frequently do you expect it to change and whether  you think each version will break the last one or if it will be more of an additive process where you add new folders to the existing structure?

Reply:
This refers to supported version of radiance folder structure:
https://github.com/ladybug-tools/radiance-folder-structure/tags

We need to update it every time that we update the module to support a new release for folder structure. At some point we need to add the version to the folder structure itself so we can load the folder version for validation. I'm still not sure how is the  best approach to do that. Also don't want to create an exception for backwards compatibility support in this stage.